src/ads-client.ts

```diff
@@ -942,10 +942,9 @@ export class Client extends EventEmitter<AdsClientEvents> {
       try {
         this.socketWrite(buffer);
       } catch (err) {
-        reject(err);
-      } finally {
         this.amsTcpCallback = undefined;
         clearTimeout(this.portRegisterTimeoutTimer);
```

Comment:
Perhaps the timeout needs to be cleared in `finally` still?

Reply:
We shouldn't clear the timeout in finally, that would cause the timeout never work. The idea is that if we aren't getting any response, we should just destroy the socket connection.

Reply:
Ah yes, of course. Makes sense.